packages/amplify-e2e-core/src/categories/api.ts

```diff
@@ -1102,6 +1102,37 @@ export const importRDSDatabase = (cwd: string, opts: ImportApiOptions & { apiExi
   });
 };
 
+export const generateUnauthSQL = (
+  cwd: string,
+  opts: { sqlSchema: string; engineType: string; out: string; expectMessage?: string },
+): Promise<void> => {
+  const options = _.assign(defaultOptions, opts);
+
+  return new Promise<void>((resolve, reject) => {
+    const generateCommand = spawn(
+      getCLIPath(options.testingWithLatestCodebase),
+      ['api', 'generate-schema', '--sql-schema', opts.sqlSchema, '--engine-type', opts.engineType, '--out', opts.out],
+      {
+        cwd,
+        stripColors: true,
+        noOutputTimeout: VPC_DEPLOYMENT_WAIT_TIME,
+      },
+    );
+    generateCommand.expect('This feature is in preview and is not recommended to use with production systems.');
+    if (opts.expectMessage) {
+      generateCommand.expect(opts.expectMessage);
+    }
+
+    generateCommand.run((err: Error) => {
+      if (!err) {
+        resolve();
+      } else {
+        reject(err);
+      }
+    });
+  });
```

Comment:
You should be able to lose the `new Promise...` and just use `return generateCommand.runAsync()` at the end, right?

Reply:
I was going to say we don't have this functionality, because it is not available in the codegen repo. It is available in this repo though 🤔